lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs

```diff
@@ -149,6 +161,52 @@ spec = do
                     shouldThrow io $ \(ErrUnexpectedNetworkFailure link _) ->
                         show link == show (safeLink api (Proxy @GetTipId))
             bracket (startNode wrongUrl wait) killNode test
+
+
+    -- NOTE: 'Right ()' just means that the format wasn't obviously wrong.
+    -- The tx may still be rejected.
+    describe "Submitting signed transactions (that are not obviously wrong)"
+        $ beforeAll startNode' $ afterAll killNode $ do
+
+        it "empty tx succeeds" $ \(_, nw) -> do
+            let signedEmpty = (Tx [] [], [])
+            runExceptT (postTx nw signedEmpty) `shouldReturn` Right ()
+
+        it "some tx succeeds" $ \(_, nw) -> do
+            let signed = (txNonEmpty, [pkWitness])
+            runExceptT (postTx nw signed) `shouldReturn` Right ()
+
+        it "more inputs than witnesses" $ \(_, nw) -> do
+            let signed = (txNonEmpty, [])
+            let err = Left $ ErrPostTxBadRequest ""
+            runExceptT (postTx nw signed) `shouldReturn` err
+
+        it "more witnesses than inputs - fine apparently" $ \(_, nw) -> do
+            -- Becase of how signed txs are encoded:
+            -- n                      :: Word8
+            -- m                      :: Word8
+            -- in_0 .. in_n           :: [TxIn]
+            -- out_0 .. out_m         :: [TxOut]
+            -- witness_0 .. witness_n :: [TxWitness]
+            --
+            -- this should in practice be like appending bytes to the end of
+            -- the message.
+            let signed = (txNonEmpty, [pkWitness, pkWitness, pkWitness])
+            runExceptT (postTx nw signed) `shouldReturn` Right ()
+
+        it "no input, one output" $ \(_, nw) -> do
+            let tx = (Tx []
+                    [ (TxOut $ unsafeDecodeAddress proxy "ca1qwunuat6snw60g99ul6qvte98fjale2k0uu5mrymylqz2ntgzs6vs386wxd")
```

Comment:
https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#limit-line-length-to-80-characters

:point_up: